### PR TITLE
Issue #908 - Surface Service/Ingress external IPs, hostname to application

### PR DESCRIPTION
**Pull request by alexmt**

### CODE REVIEW

File: pkg/apis/application/v1alpha1/generated.proto [hunk 1]
```diff
@@ -225,8 +226,11 @@ message ApplicationStatus {
   optional k8s.io.apimachinery.pkg.apis.meta.v1.Time observedAt = 8;
 
   optional string sourceType = 9;
+
+  repeated k8s.io.api.core.v1.LoadBalancerIngress ingress = 10;
 }
 
+// ApplicationTree holds nodes which belongs to the application
```
Comment (alexec) [inline pkg/apis/application/v1alpha1/generated.proto]: 👍

File: pkg/apis/application/v1alpha1/generated.proto [hunk 2]
```diff
@@ -496,12 +500,15 @@ message ResourceIgnoreDifferences {
...
 
+// ResourceNetworkingInfo holds networking resource related information
 message ResourceNetworkingInfo {
   map<string, string> targetLabels = 1;
 
   repeated ResourceRef targetRef = 2;
 
   map<string, string> labels = 3;
+
+  repeated k8s.io.api.core.v1.LoadBalancerIngress ingress = 4;
```
Comment (alexec) [inline pkg/apis/application/v1alpha1/generated.proto]: Minor - do we need to prefix this is `LoadBalancer`. I feel like `Ingress` does not need to be an LB.

**Review by alexec — APPROVED** [review]
This looks neat. Do you plan to surface in the UI?

File: pkg/apis/application/v1alpha1/generated.proto (hunk 2 above)
Comment (alexmt) [reply pkg/apis/application/v1alpha1/generated.proto]: I've reused type from `core/v1` kubernetes package. For some reason they decided to use `LoadBalancerIngress` type name

**Comment by alexmt** [comment]
> This looks neat. Do you plan to surface in the UI?

Yes, here is PR: https://github.com/argoproj/argo-cd-ui/pull/99

**Pull request merged**